### PR TITLE
[tests-only] Refactor & add unit tests for app bar components

### PR DESCRIPTION
**Pull request by JammingBen**
## Description

* `CreateAndUpload.vue` - 22% -> **82%**
* `CreateSpace.vue` - 43% -> **100%**
* `SharesNavigation.vue` - 0% -> **100%**
* `SidebarToggle.vue` - 0% -> **100%**
* `ViewOptions.vue` - 89% -> **100%**

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests

### CODE REVIEW

File: packages/web-app-files/tests/unit/components/AppBar/SharesNavigation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+describe('SharesNavigation component', () => {
+  it('renders a shares navigation for both mobile and a desktop viewports', () => {
+    const { wrapper } = getWrapper()
+    expect(wrapper).toMatchSnapshot()
+  })
+})
+
+function getWrapper({ currentRouteName = locationSharesWithMe.name } = {}) {
+  const storeOptions = { ...defaultStoreMockOptions }
+  const store = new Vuex.Store(defaultStoreMockOptions)
```
Comment (dschmidt) [inline packages/web-app-files/tests/unit/components/AppBar/SharesNavigation.spec.ts:36]: You're not using the storeOptions defined above

File: packages/web-app-files/tests/unit/components/AppBar/ViewOptions.spec.js [hunk 2]
```diff
@@ -1,100 +0,0 @@
...
-
-    expect(select.exists()).toBe(true)
-
-    select.vm.$emit('input', 500)
-    await wrapper.vm.$nextTick()
-    expect(window.localStorage.getItem('oc_options_items-per-page')).toBe('500')
-
-    select.vm.$emit('input', 'all')
-    await wrapper.vm.$nextTick()
-    expect(window.localStorage.getItem('oc_options_items-per-page')).toBe('all')
```
Comment (dschmidt) [inline packages/web-app-files/tests/unit/components/AppBar/ViewOptions.spec.js:59]: 🙈 👀

**Review by dschmidt — APPROVED** [review]
pardon my french: GEILER SCHEISS

**Review by kulmann — APPROVED** [review]
good stuff 😎

**Pull request merged**